docs/stratum.md

```diff
@@ -0,0 +1,123 @@
+AESTRATUM application (alpha)
+==========
+
+Introduction
+==========
+
+By setting of the configuration value `stratum -> enabled` to `true`, the node also starts the aestratum app.
+
+Aestratum allows users to form mining pools where they act as a pool operator, specifying the percentage of the reward they take from mining reward for providing the pool to individual miners.
+An Aestratum client [https://github.com/aeternity/aestratum_client] connects to operator pool node (Stratum server), and solves cryptographic puzzles.
+The server sends jobs to the clients, they try to solve them and send their results (called shares) back. The server keeps targets for each client in line with its computational power and distributes rewards.
+When miner finds the result artefacts of the cuckoo cycle algorithm - nonce and proof of work, the miner is awarded a share in a mining epoch happening between two subsequent key blocks of the chain.
+A share is understood as a smallest unit of eligible right to be paid out if the share was created in the mining epoch to be rewarded.
+If the difficulty of the solution matches the difficulty of the main network, we can combine the block candidate with the computed nonce and proof of work, construct a next block and publish it.
+
+If the network accepts our new block, the node which constructed that next block will receive a mining reward. This reward will be used for paying out of the participants which submitted shares during mining epochs.
+When straum app notices that the latest keyblock changed, stratum looks back 180 epochs behind top keyblock and locates two (configurable) subsequent mining epochs.
```

Comment:
I guess `180` magic number is `BENEFICIARY_REWARD_DELAY` - it's worth mentioning that.

Reply:
added, although this is not configurable as we want to ensure that mined blocks are not reversed